src/libraries/System.Net.NameResolution/src/System/Net/NameResolutionMetrics.cs

```diff
@@ -19,19 +19,29 @@ internal static class NameResolutionMetrics
 
         public static bool IsEnabled() => s_lookupDuration.Enabled;
 
-        public static void AfterResolution(TimeSpan duration, string hostName, string? errorType)
+        public static void AfterResolution(TimeSpan duration, string hostName, Exception? exception)
         {
             var hostNameTag = KeyValuePair.Create("dns.question.name", (object?)hostName);
 
-            if (errorType is null)
+            if (exception is null)
             {
                 s_lookupDuration.Record(duration.TotalSeconds, hostNameTag);
             }
             else
             {
-                var errorTypeTag = KeyValuePair.Create("error.type", (object?)errorType);
+                var errorTypeTag = KeyValuePair.Create("error.type", (object?)GetErrorType(exception));
                 s_lookupDuration.Record(duration.TotalSeconds, hostNameTag, errorTypeTag);
             }
         }
+
+        private static string GetErrorType(Exception exception) => (exception as SocketException)?.SocketErrorCode switch
+        {
+            SocketError.HostNotFound => "host_not_found",
+            SocketError.TryAgain => "try_again",
+            SocketError.AddressFamilyNotSupported => "address_family_not_supported",
+            SocketError.NoRecovery => "no_recovery",
+
+            _ => exception.GetType().Name
```

Comment:
Full name is used in other tags like this.

Reply:
Good catch! This is an overlook, [Liudmila's spec](https://github.com/lmolkova/semantic-conventions/blob/dotnet8-metrics/docs/dotnet/dotnet-http-metrics.md#metric-httpclientrequestduration) on `http.client.request.duration` requires us to use the full name:

> When error.type attribute is reported, it contains one of [HTTP Request errors](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestError.cs), a full exception type, or a string representation of received status code.

We should update this for consistence (with the spec and aspnet) as part of the sync-up effort we are doing now, I will add the change of the HTTP tag value to #93255.

Reply:
Was the intention behind "full exception type" actually the full .NET type name, or more so along the lines of "use HttpRequestException instead of just 'Http' or 'HttpException'"?

IMO `Name` feels appropriate here, I don't think `System.Threading.Tasks.TaskCanceledException` buys us anything over `TaskCanceledException`.

Reply:
I tend to agree. Will not touch the existing behavior in the PRs, opened https://github.com/dotnet/runtime/issues/93302.